File: Algorithms/s_bubble.py
# The bubble sort makes multiple passes through a list. 

# It compares adjacent items and exchanges those that are out of order.
# Each pass through the list places the next largest value in its proper place. 

# ------------------------------------
# I made the first algo when i was learning python; the one with the function is better to use

# lst = [919, 15, 1, 4, 56, 32, 25, 85, 10, 3]
# needSorting = True
# i2 = 0

# # With a flag
# while needSorting :
#     needSorting = False  
#     for c in range (0, len(lst)):
#         for i2 in range (0, len(lst) - 1 - c): # a sorted number is no more checked
#             if lst[i2] > lst[i2+1]:                    
#                 lst[i2], lst[i2+1] = lst[i2+1], lst[i2] #bubble'd
#                 needSorting = True                          
#                 print("sorting in progress -> {}".format(lst)) # numbers moving the their right place
        
# print("Sorted list : {}".format(lst))

# ------------------------------------
# FUNCTION (2019/04/12)
def bubbleSort(tab):
    for i in range(0, len(tab) - 1):   
        for j in range(0, len(tab) - 1 - i): # a sorted number is no more checked (i)
            if tab[j + 1] < tab[j]:
                tab[j+1], tab[j] = tab[j], tab[j+1]
    return tab

# exemple of use
print(bubbleSort([919, 15, 1, 4, 56, 32, 25, 85, 10, 3]))
print(bubbleSort([50, 5, 16]))

File: Algorithms/s_bubble.js
// The bubble sort makes multiple passes through a list. 

// It compares adjacent items and exchanges those that are out of order.
// Each pass through the list places the next largest value in its proper place. 

// ------------------------------------
// I made this file when i was learning js; will convert it to a function later

const arr = [15, 1, 53, 5, 56, 32, 25, 919, 85, 6];
let isChanged = true; // flag is used to stop if the array is already sorted ; no need to go till the end of array

while (isChanged) {
    isChanged = false
    for ( let i = 0; i <= arr.length; i++) {
        for ( let j = 0; j < (arr.length - 1 - i); j++) { // a sorted number is no more sorted
            if (arr[j] > arr[j+1]) {
                const tempStock = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = tempStock;
                isChanged = true;
            }
        console.log("array in progress -> " + arr);
        }
    }
}
console.log("final array sorted -> " + arr)

// 2019/04/26
// BELLOC Boris
